estate/models/estate_offer.py

```diff
@@ -0,0 +1,63 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+
+
+class EstateOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "property offer"
+
+    price = fields.Float(string='Price')
+    partner_id = fields.Many2one('res.partner', required=True, string='Partner')
+    property_id = fields.Many2one('estate.property', required=True, string='Property')
+    date_creation = fields.Date(readonly=True, default=fields.Date.today)
+    status = fields.Selection(
+        string='Status',
+        copy=False,
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused')],
+    )
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        string='Deadline', compute="_compute_deadline", inverse="_inverse_deadline"
+    )
+
+    _check_price = models.Constraint(
+        'CHECK(price >= 0)', 'The offer price must be positive'
+    )
+
+    @api.depends('validity')
+    def _compute_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.add(
+                fields.Date.today(), days=record.validity
+            )
+
+    @api.onchange('date_deadline')
+    def _inverse_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - fields.Date.today()).days
+
+    def action_accept(self):
+        for record in self:
+            for offers in record.property_id.offer_ids:
+                offers.status = "refused"
```

Comment:
You can actually do this since you're setting all records of the recordset to the same value.
`record.property_id.offer_ids.status = "refused"`

And take care of the plural misused:
```suggestion
            for offer in record.property_id.offer_ids:
                offer.status = "refused"
```